packages/node-build-scripts/generate-css-variables.js

```diff
@@ -33,94 +33,139 @@ function main() {
         }).argv;
 
     const outputFilename = args["outputFileName"];
-    const scssVariables = generateScssVariables(args["_"], outputFilename, args["retainDefault"]);
-    generateLessVariables(scssVariables, outputFilename);
+    const parsedInput = getParsedVars(args["_"], args["retainDefault"]);
+    generateScssVariables(parsedInput, outputFilename);
+    generateLessVariables(parsedInput, outputFilename);
 }
 
 /**
  * Pulls together variables from the specified Sass source files, sanitizes them for consumption,
- * and writes to an output file.
+ * and gets compiled output from `get-sass-vars`.
  *
  * @param {string[]} inputSources
- * @param {string} outputFilename
  * @param {boolean} retainDefault
- * @returns {string} output Sass contents
+ * @returns {{parsedVars: object, varsInBlocks: Set<string>[]}} output compiled variable values and grouped variable blocks
  */
-function generateScssVariables(inputSources, outputFilename, retainDefault) {
+function getParsedVars(inputSources, retainDefault) {
     // concatenate sources
-    let variablesScss = inputSources.reduce((str, currentFilename) => {
+    let cleanedInput = inputSources.reduce((str, currentFilename) => {
         return str + fs.readFileSync(`${SRC_DIR}/${currentFilename}`).toString();
     }, "");
-
     // strip comments, clean up for consumption
-    variablesScss = stripCssComments(variablesScss);
-    variablesScss = variablesScss
+    cleanedInput = stripCssComments(cleanedInput);
+    cleanedInput = cleanedInput
         .replace(/(@import|\/\/).*\n+/g, "")
         .replace(/@use "sass:math";\n/g, "")
         .replace(/border-shadow\((.+)\)/g, "0 0 0 1px rgba($black, $1)")
         .replace(/\n{3,}/g, "\n\n");
     if (!retainDefault) {
-        variablesScss = variablesScss.replace(/\ \!default/g, "");
+        cleanedInput = cleanedInput.replace(/\ \!default/g, "");
     }
-    variablesScss = [COPYRIGHT_HEADER, USE_MATH_RULE, variablesScss].join("\n");
+    cleanedInput = [USE_MATH_RULE, cleanedInput].join("\n");
```

Comment:
no need for the USE_MATH_RULE now, we can remove that constant completely

Reply:
It is actually needed here so it can properly run through `getSassVars` (which compiles with `sass`, so without it, it wouldn't know how to resolve `math.div`s). When it goes through `generateScssVariables` there's no outputted `@use "sass:math"`

Reply:
oh wait, I guess we need this to compute the values... sorry, disregard